isabelle/Doc/Specification/Guarantees.thy

```diff
@@ -14,132 +14,187 @@ begin
 
 chapter \<open>Marlowe Guarantees \label{sec:marlowe-guarantees} \<close>
 
-text \<open>We can also use the Isabelle proof assistant to prove that the Marlowe semantics presents certain
-desirable properties, such as that assets are preserved and anything unspent is returned to users by
-the end of the execution of any contract.\<close>
+text \<open>We can also use the Isabelle proof assistant to prove that the Marlowe semantics
+ presents certain desirable properties, such as that assets are preserved and anything
+ unspent is returned to users by the end of the execution of any contract.\<close>
 
-section \<open>Assets Preservation\<close>
+text \<open>To keep this section concise, whenever we introduce a helper function we'll 
+provide a brief explanation and the type signature. We'll also add the name of the 
+theories and files to access the full definitions.\<close>
+
+section \<open>Positive Accounts\<close>
+
+text \<open>The Marlowe State represents the \<^emph>\<open>accounts\<close>, \<^emph>\<open>choices\<close> and \<^emph>\<open>boundValues\<close> Maps as
+associative lists \secref{sec:state}. While we make proofs in Isabelle, we often need to
+ assert that two Maps are equal, and with the list representation, this means that the 
+maps should be in the same order and without duplicate keys. To ensure this, we have the following helper located 
+in the MList.thy theory\<close>
```

Comment:
No need to expand on this, once we move to FMap we wont need a `valid_map` function at all

Reply:
If we are talking about `valid_map`, then we must do it accurately, otherwise why are we talking about `valid_map`? Also moving to FMap is not a priority

Reply:
The current version is accurate and concise, what you suggest is accurate and more in-depth. The only reason that I talk about `valid_map` is because I need to describe `validAndPositive_state` which is the subject of section as it is both in the premise and conclusion of the presented theorem. 

The only part of `validAndPositive_state` that I consider meaningful is the `Positive` part. As I mentioned both in the document and in the comment, the `valid_map` part is only necessary because of a Isabelle artifact, or a decision that we previously made around how to model the Maps. 

Even if moving to FMap is not a priority I wouldn't try to justify its existence more than the bare minimum, as it switches the focus of the important part.